stock_cancel/stock.py

```diff
@@ -0,0 +1,61 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2012 Andrea Cometa All Rights Reserved.
+#                       www.andreacometa.it
+#                       openerp@andreacometa.it
+#    Copyright (C) 2013 Agile Business Group sagl (<http://www.agilebg.com>)
+#    Ported to new API by Alexis de Lattre <alexis.delattre@akretion.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, api, _, workflow, exceptions
+
+
+class stock_picking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.model
+    def has_valuation_moves(self, move):
```

Comment:
:hand: this is a method that works on pickings. I think it should be an instance method on pickings, so:

- `@api.multi`, `ensure_one`
- no parameters
- `self.name` instead of `move.picking_id.name`
- be called as `line.picking_id.has_valuation_moves()`

Or alternatively, it can be an instance method on the move and be called as `line.has_valuation_moves()` - whichever makes more sense functionally.